lib/GADS/Schema/Result/Report.pm

```diff
@@ -220,18 +220,32 @@ has data => (
 
         my $record_id = $self->record_id;
         $record->find_current_id($record_id);
+        my $allowed_cols = [map {$_->name} $gads_layout->all(user_can_read => 1)];
 
-        while ( my $layout = $layouts->next ) {
+         while ( my $layout = $layouts->next ) {
+            next unless _is_column_allowed($self, $layout->layout->name, $allowed_cols);
             my $column = $gads_layout->column( $layout->layout_id );
```

Comment:
Thanks Dave. A simpler fix might be just to change the above line to:
`my $column = $gads_layout->column( $layout->layout_id, read => 1 ) or next`
Can you try that please?

Reply:
(also, skipping rather than showing blank would make it consistent with other parts of the app)

Reply:
Doesn't work - can now read values I'm not allowed to

Reply:
And it does skip :smile:

Reply:
Fixed this using

`my $column = $gads_layout->column( $layout->layout_id, permission=>'read' );`